diff --git a/projects/ad9361/src/main.c b/projects/ad9361/src/main.c
index 8b224295..6434b6a9 100644
--- a/projects/ad9361/src/main.c
+++ b/projects/ad9361/src/main.c
@@ -81,6 +81,8 @@
 
 #endif // IIO_SUPPORT
 
+#include <genode_backend.h>
+
 /******************************************************************************/
 /************************ Variables Definitions *******************************/
 /******************************************************************************/
@@ -127,6 +129,11 @@ struct xil_gpio_init_param xil_gpio_param = {
 #define SPI_PARAM	NULL
 #endif
 
+#define GPIO_OPS   &genode_gpio_ops
+#define SPI_OPS    &genode_spi_ops
+#define GPIO_PARAM NULL
+#define SPI_PARAM  NULL
+
 struct axi_adc_init rx_adc_init = {
 	"cf-ad9361-lpc",
 	RX_CORE_BASEADDR,
@@ -494,6 +501,13 @@ struct ad9361_rf_phy *ad9361_phy_b;
 int main(void)
 {
 	int32_t status;
+
+	/* call functions from genode_backend.h to get device addresses */
+	rx_adc_init.base  = genode_dev_addr(rx_adc_init.name);
+	tx_dac_init.base  = genode_dev_addr(tx_dac_init.name);
+	rx_dmac_init.base = genode_dev_addr(rx_dmac_init.name);
+	tx_dmac_init.base = genode_dev_addr(tx_dmac_init.name);
+
 #ifdef XILINX_PLATFORM
 	Xil_ICacheEnable();
 	Xil_DCacheEnable();
@@ -562,8 +576,10 @@ int main(void)
 	default_init_param.rx_synthesizer_frequency_hz = 2300000000UL;
 	default_init_param.tx_synthesizer_frequency_hz = 2300000000UL;
 
+#if 0
 	rx_adc_init.base = AD9361_RX_1_BASEADDR;
 	tx_dac_init.base = AD9361_TX_1_BASEADDR;
+#endif
 
 	ad9361_init(&ad9361_phy_b, &default_init_param);
 
-- 
2.33.1

